### ISSUE TITLE
Possibility to add a sorting option?

### ISSUE THREAD

**Issue by GurvanKervern** [transmission/transmission#3822]
Hi,

It would be very useful to add a --sort option to transmission-remote (or other clients), especially on dates added, created, etc, since all IDs get scrambled upon every reboot, making it more difficult to find recent torrents when you have more than a few.

Alternatively having the torrents keep the same IDs upon reboot would fix that problem as well, but sorting could be much more powerful (by size, name, ratios, etc)

**Comment by GurvanKervern** [comment]
Hmm, I'm not sure I understand : 

-t <torrent_id> -i contains the dates/times it was created, added, finished, last active, etc, so this information is stored

- when transmission-daemon adds a torrent, it copies the file into a folder (forgot which) and will reload from that folder upon next start, I suppose that's how it keeps the data from one time to the next, so the change/modify stats could be used to load them "sequentially", right?

**Comment by LaserEyess** [comment]
It does, but consider 2 torrents, torrent A and torrent B, and the following steps:

1. add torrent A
2. add torrent B
3. `transmission-remote -t 1 -i`, you get info for torrent A
4. remove torrent A
5. reboot
6. `transmission-remote -t 1 -i`

Should step 6 error out because A is gone, or should it return B?

**Comment by GurvanKervern** [comment]
ho I see your point :p 

What about the various datetimes related to the torrent, couldn't they be used to keep the same relative order? (ls -tr <folder>/* | xargs -i add_command {}) Or used to sort the torrents on printing?

**Comment by LaserEyess** [comment]
Yes, in theory they could. Your feature request for something like `--sort` may be a good idea. However it seems pretty complex. Recently this commit was merged https://github.com/transmission/transmission/commit/fe9aba702e255802b99078c69fc93b8392bc65b8, which means you can dump the response to json, then further process with jq/sort quite easily. A bit of a hack, yes, but it will work. 

Furthermore `-t` can use sha1 hashes so that will certainly be deterministic across boots. For example, if torrent B is `ae4f281df5a5d0ff3cad6371f76d5c29b6d953ec` then you can do `transmission-remote -t ae4f281df5a5d0ff3cad6371f76d5c29b6d953ec -i`.  Doing this by hand, of course, is a bit tedious, but scripting it should be fairly simple.

**Comment by 1993Banik** [comment]
I couldn't agree more with this suggestion! The ability to add a --sort option to transmission-remote is long overdue and would significantly enhance the usability of the client, especially for users who deal with a substantial number of torrents.

I wholeheartedly support this idea, and I'm open a GitHub user account to express my concerns. Your dedication to improving the software is commendable.

Incorporating a --sort option, and even better, allowing users to customize their torrent sorting preferences through settings.json, would be a fantastic addition to Transmission. It would provide users with greater control over their torrent management and make it easier to locate and manage torrents, particularly after system reboots.

Let's keep our fingers crossed that the developers take notice of this valuable feedback and prioritize it in their future releases. With each new version, we can hope for improvements that will make Transmission even more efficient and user-friendly.

Thank you for your proactive approach to addressing this issue and for sharing your thoughts with the community. Your efforts are sure to make a positive impact on the development of Transmission!

**Comment by Galicarnax** [comment]
Adding to the feature request to sort transfers by timestamps. Otherwise, with hundreds of transfers, listing is quite awkward. Grep could help, but more often than not, I can't recall immediately titles of torrents I added lately.

**Comment by GurvanKervern** [comment]
Agreed, it doesn't even need to do complicated sorts at first, just allow to show the date or timestamp(s) in the outputs like the `-l` option and we can sort/grep manually ourselves.

**Comment by SNBogdanov** [comment]
Adding sort for torrents as added time is not well documented and is bad surprise for me. After that if there is no option to put sorting back to sort by ID is even worse.
Pls. Reverse changes or add option to work as before
